File: projects/JavaFXApps/core/src/main/java/net/rickiekarp/core/settings/LoadSave.java
package net.rickiekarp.core.settings;

import net.rickiekarp.core.controller.LanguageController;
import javafx.geometry.Side;
import javafx.scene.paint.Color;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/** Annotation for indicating load/save a field. */
@Retention(RetentionPolicy.RUNTIME)
public @interface LoadSave {

    /** settings defaults **/
    String host = "https://app.rickiekarp.net/";
    int updateChannel = 0;
    int language = LanguageController.INSTANCE.getLocale();
    int themeState = 0;
    int colorScheme = 0;
    boolean animations = true;
    boolean useSystemBorders = false;
    boolean logState = false;
    boolean showTrayIcon = false;

    /** advanced settings defaults **/
    boolean debugState = false;
    Color decorationColor = Color.valueOf("#1d1d1d");
    Color shadowColorFocused = Color.valueOf("#000000");
    Color shadowColorNotFocused = Color.valueOf("#a9a9a9");
    Side tabPosition = Side.LEFT;
}

File: projects/JavaFXApps/core/src/main/java/net/rickiekarp/core/settings/LoadSave.kt
package net.rickiekarp.core.settings

import net.rickiekarp.core.controller.LanguageController
import javafx.geometry.Side
import javafx.scene.paint.Color

/** Annotation for indicating load/save a field.  */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation class LoadSave {
    companion object {
        /** settings defaults  */
        @JvmField val host = "https://api.rickiekarp.net/"
        @JvmField val updateChannel = 0
        @JvmField val language = LanguageController.locale
        @JvmField val themeState = 0
        @JvmField val colorScheme = 0
        @JvmField val animations = true
        @JvmField val useSystemBorders = true
        @JvmField val logState = false
        @JvmField val showTrayIcon = false

        /** advanced settings defaults  */
        @JvmField val debugState = false
        @JvmField val decorationColor = Color.valueOf("#1d1d1d")
        @JvmField val shadowColorFocused = Color.valueOf("#000000")
        @JvmField val shadowColorNotFocused = Color.valueOf("#a9a9a9")
        @JvmField val tabPosition = Side.LEFT
    }
}
